File: csharp/solution/Webshop.UnitTests/PasswordValidatorTests.cs
using NUnit.Framework;
using Webshop.Service.Common;

namespace Webshop.UnitTests;

[TestFixture]
public class PasswordValidatorTests
{
    [Test]
    public void IsValidPassword_WhenPasswordIsEmpty_ShouldReturnFalse()
    {
        var passwordValidator = new PasswordValidator();
        var isValid = passwordValidator.IsValidPassword("");
        Assert.That(isValid, Is.False);
    }

    [Test]
    public void IsValidPassword_WhenPasswordIsTooShort_ShouldReturnFalse()
    {
        var passwordValidator = new PasswordValidator();
        var isValid = passwordValidator.IsValidPassword("pwd123");
        Assert.That(isValid, Is.False);
    }

    [Test]
    public void IsValidPassword_WhenPasswordIsMissingCharacters_ShouldReturnFalse()
    {
        var passwordValidator = new PasswordValidator();
        var isValid = passwordValidator.IsValidPassword("12345678");
        Assert.That(isValid, Is.False);
    }

    [Test]
    public void IsValidPassword_WhenPasswordIsMissingNumbers_ShouldReturnFalse()
    {
        var passwordValidator = new PasswordValidator();
        var isValid = passwordValidator.IsValidPassword("password");
        Assert.That(isValid, Is.False);
    }

    [Test]
    public void IsValidPassword_WhenPasswordIsValid_ShouldReturnTrue()
    {
        var passwordValidator = new PasswordValidator();
        var isValid = passwordValidator.IsValidPassword("password123");
        Assert.That(isValid, Is.True);
    }
}

File: java/solution/src/test/java/Webshop/UnitTests/PasswordValidatorTests.java
package Webshop.UnitTests;

import org.junit.jupiter.api.Test;
import Webshop.Service.Common.PasswordValidator;

import static org.junit.jupiter.api.Assertions.*;

public class PasswordValidatorTests {
    @Test
    public void isValidPassword_whenPasswordIsEmpty_shouldReturnFalse() {
        PasswordValidator passwordValidator = new PasswordValidator();
        assertFalse(passwordValidator.isValidPassword(""));
    }

    @Test
    public void isValidPassword_whenPasswordIsTooShort_shouldReturnFalse() {
        PasswordValidator passwordValidator = new PasswordValidator();
        assertFalse(passwordValidator.isValidPassword("pwd123"));
    }

    @Test
    public void isValidPassword_whenPasswordIsMissingCharacters_shouldReturnFalse() {
        PasswordValidator passwordValidator = new PasswordValidator();
        assertFalse(passwordValidator.isValidPassword("12345678"));
    }

    @Test
    public void isValidPassword_whenPasswordIsMissingNumbers_shouldReturnFalse() {
        PasswordValidator passwordValidator = new PasswordValidator();
        assertFalse(passwordValidator.isValidPassword("password"));
    }

    @Test
    public void isValidPassword_whenPasswordIsValid_shouldReturnTrue() {
        PasswordValidator passwordValidator = new PasswordValidator();
        assertTrue(passwordValidator.isValidPassword("password123"));
    }
}


// mvn -Dtest=PasswordValidatorTest test